squid-gen-targets/src/parquet/index.ts

```diff
@@ -0,0 +1,170 @@
+                    let repetitionIndex = overlaps[fieldName]
+                    if (repetitionIndex == null) {
+                        let ols = fragment.params.filter((p) => toFieldName(p.name) === fieldName)
+                        if (ols.length > 1) {
+                            repetitionIndex = overlaps[fieldName] = 0
+                        } else if (ols.length > 0 && fieldName !== toFieldName(param.name)) {
+                            repetitionIndex = overlaps[fieldName] = 0
+                        } else {
+                            break
+                        }
+                    }
+
+                    if (param.static) {
+                        break
+                    } else {
+                        overlaps[fieldName] += 1
+                        fieldName += repetitionIndex
+                    }
+                }
+                fields.push({
+                    name: fieldName,
+                    type: toParquetType(param.type),
+                    nullable: param.nullable ?? false,
+                })
+            }
+            this.tableMap.set(fragment, {name: tableName, alias: toAliasName(tableName), fields})
+        }
+    }
+
+    async generate() {
+        let tables = this.src.file('table.ts')
+        tables.line(`import {Table, Types, Column} from '@subsquid/file-store-parquet'`)
+        for (let t of this.tableMap.values()) {
+            tables.line()
+            tables.line(`export let ${t.alias} = new Table(`)
+            tables.indentation(() => {
+                tables.line(`'${t.name}.parquet',`)
+                tables.block(``, () => {
+                    for (let field of t.fields) {
+                        tables.line(`${field.name}: Column(Types.${field.type}, {nullable: ${field.nullable}}),`)
+                    }
+                })
+            })
+            tables.line(`)`)
+        }
+        tables.write()
+
+        let db = this.src.file('db.ts')
+        let destType: 'local' | 's3'
+        db.lazy(() => {
+            db.line(`import {Store as Store_, Database} from '@subsquid/file-store'`)
+            switch (destType) {
+                case 'local':
+                    db.line(`import {LocalDest} from '@subsquid/file-store'`)
+                    break
+                case 's3':
+                    db.line(`import {S3Dest} from '@subsquid/file-store-s3'`)
+            }
+            db.line(`import * as tables from './table'`)
+        })
+        db.line()
+        db.line(`export let db = new Database({`)
+        db.indentation(() => {
+            db.line(`tables,`)
+
+            let url = parseUrl(this.options.path)
+            if (url == null) {
+                destType = 'local'
+                db.line(`dest: new LocalDest('${this.options.path}'),`)
+            } else {
+                assert(url.protocol == 's3:')
+                destType = 's3'
+                db.line(`dest: new S3Dest('${url.pathname}', '${url.host}'),`)
```

Comment:
IDK about Amazon, but Filebase API does not process folder names starting with `/` in any reasonable way. I tried it and it created a weird nameless folder that I couldn't access. I suggest removing any leading slashes:
``db.line(`dest: new S3Dest('${url.pathname.replace(/^\/+/g, "")}', '${url.host}'),`)``